packages/core/src/change-matrix.ts

```diff
@@ -9,61 +17,105 @@ export interface IOnboardingChangeMatrixSpec extends IOnboardingSpec {
   yAxis?: ISpecProp;
 }
 
+function generateMessages(
+  spec: IOnboardingChangeMatrixSpec,
+  visElement: Element
+): IOnboardingMessage[] {
+  const reading = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.READING
+  ) as IOnboardingStage;
+  let messages: IOnboardingMessage[];
 
-function generateMessages(spec: IOnboardingChangeMatrixSpec, visElement: Element): IOnboardingMessage[] {
-  const reading = defaultOnboardingStages.get(EDefaultOnboardingStages.READING) as IOnboardingStage;
-  const messages = [
-    {
-      anchor: getAnchor(spec.chartTitle, visElement),
-      requires: ['chartTitle'],
-      text: `The <span class="visahoi-tooltip-hover-text">chart</span> shows the ${spec.chartTitle?.value}.`,
-      title: 'Reading the chart',
-      onboardingStage: reading,
-      tooltipPosition: "top" as TooltipPosition, // this causes the "jumping" of the tooltip when resizing
-      marker: {
-        radius: 8,
-        content: "",
-        fontSize: '20px',
-        id: "unique-marker-id-1"
-      }
-    },
-    {
-      anchor: getAnchor(spec.type, visElement),
-      requires: ['type'],
-      text: `The chart Is based on colored ${spec.type?.value} elements.`,
-      title: 'Reading the chart',
-      onboardingStage: reading,
-      tooltipPosition: "left" as TooltipPosition,
-      marker: {
-        id: "unique-marker-id-2"
-      }
-    },
-    {
-      anchor: getAnchor(spec.legendTitle, visElement),
-      requires: ['legendTitle'],
-      text: `The legend shows the ${spec.legendTitle?.value} for the chart. The colors range from blue to white and brown.`,
-      title: 'Reading the chart',
-      onboardingStage: reading,
-      marker: {
-        id: "unique-marker-id-3"
-      }
-    },
-    {
-      anchor: getAnchor(spec.xAxis, visElement),
-      requires: ['xAxis', 'yAxis'],
-      text: `The columns show the ${spec.xAxis?.value}, while the rows show the ${spec.yAxis?.value}.`,
-      title: 'Reading the chart',
-      onboardingStage: reading,
-      marker: {
-        id: "unique-marker-id-4"
-      }
-    },
-  ];
+  if (spec.chartTitle?.value === undefined) {
+    messages = [
```

Comment:
same here. please create the array with common messages outside the if and only push the other messages to this array inside the if